### PR TITLE
docs: add `Can't modify immutable headers.` tips to proxy example

### PR DESCRIPTION
**Pull request by sor4chi**
https://github.com/honojs/hono/issues/1572

スクリーンショット 2024-01-15 午前11 54 43

### CODE REVIEW

File: guides/examples.md [hunk 1]
```diff
@@ -78,6 +78,21 @@ app.get('*', (c) => {
...
+```ts
+app.get('/', async (_c) => {
+  const response = await fetch('https://example.com')
+  // clone the response to return a response with modifiable headers
+  const newResponse = new Response(response.body, response)
+  return newResponse
+})
+```
+
+Since `fetch()` returns unalterable headers, any subsequent attempt to rewrite them with `cors` middleware or the like will result in an error.
```
Comment (yusukebe) [inline guides/examples.md:93]: Rewrite like the following:

"The headers of `Response` returned by `fetch` are immutable. So, an error will occur if you modify it."

**Comment by yusukebe** [comment]
Hi @sor4chi 

Please check the comment.

**Comment by sor4chi** [comment]
@yusukebe fixed!

File: guides/examples.md [hunk 2]
```diff
@@ -78,6 +78,21 @@ app.get('*', (c) => {
...
+```ts
+app.get('/', async (_c) => {
+  const response = await fetch('https://example.com')
+  // clone the response to return a response with modifiable headers
+  const newResponse = new Response(response.body, response)
+  return newResponse
+})
+```
+
+The headers of Response returned by fetch are immutable. So, an error will occur if you modify it.
```
Comment (yusukebe) [inline guides/examples.md:93]: Please make `Response` and `fetch` as code spans.

Comment (sor4chi) [reply guides/examples.md]: opps, sorry!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Pull request merged**